src/contract/aci.ts

```diff
@@ -0,0 +1,495 @@
+  topics: Array<string | number>
+}
+
+interface DecodedEvent {
+  name: string
+  args: unknown
+  contract: {
+    name: string
+    address: string
+  }
+}
+
+interface TxData {
+  blockHash: string
+  blockHeight: number
+  hash: string
+  signatures: any[]
+  tx: object[]
+  rawTx: string
+  callerId: string
+  callerNonce: number
+  contractId: string
+  gasPrice: number
+  gasUsed: number
+  height: number
+  log: any[]
+  returnType: string
+  returnValue: string
+}
+
+interface ContractInstance {
+  _aci: Aci
+  _name: string
+  calldata: any
+  source?: string
+  bytecode?: string
+  deployInfo: {
+    address?: string
+    result?: {
+      callerId: string
+      callerNonce: number
+      contractId: string
+      gasPrice: number
+      gasUsed: number
+      height: number
+      log: any[]
+      returnType: string
+      returnValue: string
+    }
+    owner?: string
+    transaction?: string
+    rawTx?: string
+    txData?: TxData
+  }
+  options: any
+  compilerVersion: string
+  compile: (options?: {}) => Promise<string>
+  _estimateGas: (name: string, params: any[], options: any) => Promise<number>
+  deploy: (params: any[], options: object) => Promise<any>
+  call: (fn: string, params?: any[], options?: {}) => Promise<any>
+  decodeEvents: (events: Event[], { omitUnknown, ...opt }: {
+    omitUnknown?: boolean}) => Array<DecodedEvent | null>
+  methods: any
+}
+
+/**
+* Generate contract ACI object with predefined js methods for contract usage - can be used for
+* creating a reference to already deployed contracts
+* @alias module:@aeternity/aepp-sdk/es/contract/aci
+* @param options Options object
+* @returns JS Contract API
+* @example
+* const contractIns = await aeSdk.getContractInstance({ source })
+* await contractIns.deploy([321]) or await contractIns.methods.init(321)
+* const callResult = await contractIns.call('setState', [123]) or
+* await contractIns.methods.setState.send(123, options)
+* const staticCallResult = await contractIns.call('setState', [123], { callStatic: true }) or
+* await contractIns.methods.setState.get(123, options)
+* Also you can call contract like: await contractIns.methods.setState(123, options)
+* Then sdk decide to make on-chain or static call(dry-run API) transaction based on function is
+* stateful or not
+*/
+export default async function getContractInstance ({
+  source,
+  bytecode,
+  aci: _aci,
+  contractAddress,
+  fileSystem = {},
+  validateBytecode,
+  ...otherOptions
+}: {
+  source?: string
+  bytecode?: string
+  aci?: Aci
+  contractAddress?: string
+  fileSystem?: object
+  validateBytecode?: boolean
+} = {}): Promise<ContractInstance> {
+  if (_aci == null && source != null) {
+    _aci = await this.compilerApi.generateACI({ code: source, options: { fileSystem } })
+  }
+  if (_aci == null) throw new MissingContractDefError()
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+  contractAddress = contractAddress && await this.resolveName(
```

Comment:
```suggestion  
contractAddress = this.resolveName(
```

This is an interesting case however it works by removing the check. Give it a try and proceed as you'd prefer with it